bin/sign

```diff
@@ -0,0 +1,97 @@
+ * package version has already been signed, we download the signed xpi
+ * from amo, and move the files into the dist/ directory.
+ * If the package version has not been signed, sign the xpi, and move
+ * them into the /dist directory.
+ *
+ */
+var fs = require('fs');
+var path = require('path');
+var request = require('request');
+var jwt = require('jsonwebtoken');
+var manifestFilename = __dirname + '/../package.json';
+var manifest = require(manifestFilename);
+var version = manifest.version;
+var apiKey = process.env['AMO_USER'];
+var apiSecret = process.env['AMO_SECRET'];
+var authToken = jwt.sign({iss: apiKey}, apiSecret, {
+  algorithm: "HS256",
+  expiresIn: 60
+});
+var signedOpts = {
+  url: 'https://addons.mozilla.org/api/v3/addons/@' + manifest.name + '/versions/'+version+ '/',
+  headers: {
+    'Authorization': 'JWT ' + authToken
+  }
+};
+
+request(signedOpts, signCb);
+
+function signCb(err, resp, body) {
+  if (!err && resp.statusCode == 200) {
+    var info = JSON.parse(body);
+    if (info.files.length) {
+      var ws = fs.createWriteStream('dist/addon-signed.xpi').on('finish', removeGeneratedXpi);
+      signedOpts.url = info.files[0].download_url;
+      request(signedOpts).pipe(ws);
+    }
+  } else distAddon();
+}
+
+// if we need to sign and distribute our add-on, we want to use this method
+function distAddon() {
+  // sign our add-on
+  var generatedXpi = 'dist/addon.xpi';
+  signAddon(generatedXpi, function(err, signedXpiPath) {
+    if (err) return console.error(err);
+    // remove our generated xpi since we now have a signed version
+    removeGeneratedXpi();
+    // move our signed xpi and rdf into the /dist dir
+    // directory and exit
+    checkExistsAndMv(signedXpiPath, 'dist/signed-addon.xpi', function(err) {
+      if (err) return console.error(err);
+      console.log('addon.xpi written to dist/signed-addon.xpi');
+    });
+    // TODO:(DJ) handle update.rdf generation
+    // var generatedRdf = '@testpilot-addon-' + version + '.update.rdf';
+    // checkExistsAndMv(generatedRdf, staticProjectPath + '/update.rdf', function(err) {
+    //   if (err) return console.error(err);
+    //   console.log('update.rdf written to ' + staticProjectPath + '/update.rdf');
+    // });
+  });
+}
+
+function removeGeneratedXpi() {
+  var generatedXpi = 'dist/addon.xpi';
+  fs.unlink(generatedXpi, function(err) {
+    if (err) console.error(err);
+    else console.log('removed ' + generatedXpi + ' successfully');
+  });
+}
+
+function signAddon(xpiPath, cb) {
+  require('jpm/lib/sign').sign({
+    apiKey: apiKey,
+    apiSecret: apiSecret,
+    xpi: xpiPath
+  }).then(function(result) {
+    if (result.success) cb(null, result.downloadedFiles[0]);
+  }).catch(cb);
+}
+
+function checkExistsAndMv(fromFilePath, toFilePath, cb) {
+  fs.stat(fromFilePath, function(err) {
+    if (err) return cb(err);
+    fs.rename(fromFilePath, toFilePath, function(err) {
+      if (err) return cb(err);
+      else cb();
+    });
+  });
+}
```

Comment:
Since this file is doing a lot of file operations, not sure if it'd be easier to use something like [**fs-extra**](https://github.com/jprichardson/node-fs-extra) and synchonous methods.